packages/techdocs-common/src/stages/prepare/commonGit.ts

```diff
@@ -31,23 +35,45 @@ export class CommonGitPreparer implements PreparerBase {
     this.logger = logger;
   }
 
-  async prepare(entity: Entity): Promise<string> {
+  async prepare(
+    entity: Entity,
+    options?: { etag?: string },
+  ): Promise<PreparerResponse> {
+    this.logger.warn(
+      'You are using the legacy git preparer in TechDocs which will be removed in near future (March 2021). ' +
+        'Migrate to URL reader by updating `backstage.io/techdocs-ref` annotation in `catalog-info.yaml` ' +
+        'to be prefixed with `url:`. Read the migration guide and benefits at https://github.com/backstage/backstage/issues/4409 ',
+    );
+
     const { target } = parseReferenceAnnotation(
       'backstage.io/techdocs-ref',
       entity,
     );
 
     try {
+      // Update repository or do a fresh clone.
       const repoPath = await checkoutGitRepository(
         target,
         this.config,
         this.logger,
       );
-      const parsedGitLocation = parseGitUrl(target);
+      // Check if etag has changed for cache invalidation.
+      const etag = await getLastCommitTimestamp(repoPath, this.logger);
+      if (options?.etag === etag.toString()) {
+        throw new NotModifiedError();
```

Comment:
Hmm do we want to throw an error when nothing is modified and its still good to use? or do we just want to do nothing? I guess we are mostly interested in an action when its changed.

Reply:
continued to review and could see it's mostly for debugging purposes, which I guess make sense

Reply:
Yep, this error is not transmitted to the user.